pkg/server/server_test.go

```diff
@@ -265,8 +265,8 @@ func baseServerProxyTestWithoutBackend(t *testing.T, validate func(*agentmock.Mo
 
 	proxyServer.Proxy(frontendConn)
 
-	// add a sleep to make sure `serveRecvFrontend` ends after `Proxy` finished.
-	time.Sleep(1 * time.Second)
+	// make sure `serveRecvFrontend` ends after `Proxy` finished.
```

Comment:
My suggestion here got too long, so I decided it was easier to just send a PR. What do you think of this approach? https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/427

Reply:
Yes that approach is definitely much better.

/hold